src/Kentico.Xperience.Lucene.Core/Indexing/DefaultLuceneTaskLogger.cs

```diff
@@ -63,9 +65,13 @@ public async Task HandleReusableItemEvent(IndexEventReusableItemModel reusableIt
 
             if (toReindex is not null)
             {
-                foreach (var item in toReindex)
+                if (taskType == LuceneTaskType.DELETE)
```

Comment:
@bkapustik @hesspavel  This change is completely breaking the indexing of the items that were provided to re-index by `FindItemsToReindex`. Root cause of https://github.com/Kentico/xperience-by-kentico-lucene/issues/151.